### DISCUSSION TITLE
LDAP ( "ОШИБКА:Operations error" )

### DISCUSSION THREAD

**Discussion by DeAlexPesh** [totumonline/totum-ru-issues-and-discussions#391] [ОШБК!] [answered]
###  **_Проблема:_**

Ошибка `"ОШИБКА:Operations error"` при обращении к корню домена: "basedn": "dc=domain,dc=ru"
#

###  **_Причина ([документация](https://www.php.net/manual/ru/function.ldap-search.php#45388)):_**

> In order to perform the searches on Windows 2003 Server Active Directory you have to set the LDAP_OPT_REFERRALS option to 0:
> ldap_set_option($ldap, LDAP_OPT_REFERRALS, 0);
> Without this, you will get "Operations error" if you try to search the whole AD schema (using root of the domain as a $base_dn).
> As opposed to Windows 2000 Server, where this option was optional and only increased the performance.
#

###  **_Решение:_**

Параметр поля `h_version`
```
{
  "LDAP_OPT_REFERRALS": "0",
  "LDAP_OPT_PROTOCOL_VERSION": "3"
}
```

**Comment by ttm-alexeypolunin** [comment, marked as answer]
**Спасибо!**

Да, ранее это решалось указанием конкретного `OU`:

![IMAGE 2023-05-24 19:54:23](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/fb2e5c8f-c0e1-4bbb-b639-dfc56f85062e)
![photo_2023-05-24 19 54 00](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/2d64f37c-a2be-42b6-a946-bba0446648d3)

![IMAGE 2023-05-24 19:54:40](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/4b8c1a2a-ef58-4361-9903-1b03ca171c85)
![photo_2023-05-24 19 54 03](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/4bbd8a66-0e49-47bc-aeef-b8ba4e70fac4)

### Уточнения

 `"LDAP_OPT_PROTOCOL_VERSION": "3"` — можно не указывать, он добавится автоматически. Если указать, то он переписывается поверх.
 
В таком виде параметры передаются в `ldap_set_option` с указанным `$ldap`:

```
ldap_set_option($ldap, LDAP_OPT_REFERRALS, 0);
```

Если нужно передать без `$ldap` то нужно указать у параметра префикс `G_`
